features/nanostack/mbed-mesh-api/source/MeshInterfaceNanostack.cpp

```diff
@@ -182,9 +184,9 @@ void Nanostack::Interface::network_handler(mesh_connection_status_t status)
         _connect_status = NSAPI_STATUS_DISCONNECTED;
     }
 
-    if (_connection_status_cb && _previous_connection_status != _connect_status
-            && (_previous_connection_status != NSAPI_STATUS_GLOBAL_UP || status != MESH_BOOTSTRAP_STARTED)
-            && (_previous_connection_status != NSAPI_STATUS_CONNECTING || status != MESH_BOOTSTRAP_START_FAILED)) {
+    if (_connection_status_cb && (global_up || (_previous_connection_status != _connect_status
```

Comment:
Could you just use `_connect_status == NSAPI_STATUS_GLOBAL_UP` instead of introducing new flag?